ironfish-cli/src/commands/miners/start.ts

```diff
@@ -11,10 +12,23 @@ export class Miner extends IronfishCommand {
 
   static flags = {
     ...RemoteFlags,
+    threads: flags.integer({
+      char: 't',
+      default: 1,
+      description:
+        'number of CPU threads to use for mining. -1 will auto-detect based on number of CPU cores.',
+    }),
   }
 
   async start(): Promise<void> {
-    this.parse(Miner)
+    const { flags } = this.parse(Miner)
+
+    let threads = flags.threads
+    if (threads === 0 || threads < -1) {
+      throw new Error('--threads must be a positive integer or -1.')
+    } else if (threads === -1) {
+      threads = (await import('os')).cpus().length
```

Comment:
is there a reason to do a lazy import here, instead of at the top?

Reply:
Changed